app/controllers/provider_interface/provider_agreements_controller.rb

```diff
@@ -0,0 +1,55 @@
+module ProviderInterface
+  class ProviderAgreementsController < ProviderInterfaceController
+    before_action :set_page_name
+    skip_before_action :check_data_sharing_agreements
+
+    def new_data_sharing_agreement
+      @provider_agreement = GetPendingDataSharingAgreementsForProviderUser.call(provider_user: current_provider_user).first
+      if @provider_agreement
+        render_agreement
+      else
+        redirect_to action: :show_data_sharing_agreement
+      end
+    end
+
+    def create_data_sharing_agreement
+      @provider_agreement = ProviderAgreement.new(provider_agreement_params.merge(provider_user: current_provider_user))
+      begin
+        @provider_agreement.save!
+        redirect_to '/provider'
+      rescue ActiveRecord::RecordInvalid
```

Comment:
When do we expect this to occur?

Reply:
If a user modifies the html to supply a `provider_id` they are not associated with.